File: core/src/com/colorflow/entity/dot/Color.java
package com.colorflow.entity.dot;

import java.util.List;

public enum Color {
    RED, YELLOW, GREEN, MAGENTA, BLUE, CYAN;

    public static Color getRandomExcept(List<Color> colors) {
        int colorNum;
        while (true) {
            colorNum = (int) (Math.random() * 6.0);
            switch (colorNum) {
                case 0:
                    if (!colors.contains(Color.RED)) {
                        return RED;
                    }
                    break;
                case 1:
                    if (!colors.contains(Color.YELLOW)) {
                        return YELLOW;
                    }
                    break;
                case 2:
                    if (!colors.contains(Color.GREEN)) {
                        return GREEN;
                    }
                    break;
                case 3:
                    if (!colors.contains(Color.MAGENTA)) {
                        return MAGENTA;
                    }
                    break;
                case 4:
                    if (!colors.contains(Color.BLUE)) {
                        return BLUE;
                    }
                    break;
                case 5:
                    if (!colors.contains(Color.CYAN)) {
                        return CYAN;
                    }
                    break;
                default:
                    throw new IllegalStateException();
            }
        }
    }

    public static float getAngleBetween(Color c1, Color c2) {
        int dif = c1.ordinal() - c2.ordinal();
        return 60 * dif;
    }

    public com.badlogic.gdx.graphics.Color getRGB() {
        switch (this) {
            case RED:
                return com.badlogic.gdx.graphics.Color.RED;
            case YELLOW:
                return com.badlogic.gdx.graphics.Color.YELLOW;
            case GREEN:
                return com.badlogic.gdx.graphics.Color.GREEN;
            case MAGENTA:
                return com.badlogic.gdx.graphics.Color.MAGENTA;
            case BLUE:
                return com.badlogic.gdx.graphics.Color.BLUE;
            case CYAN:
                return com.badlogic.gdx.graphics.Color.CYAN;
            default:
                throw new IllegalStateException();
        }
    }
}

File: core/src/com/colorflow/play/entity/dot/Color.kt
package com.colorflow.play.entity.dot

enum class Color {
    RED, YELLOW, GREEN, MAGENTA, BLUE, CYAN;

    val rgb: com.badlogic.gdx.graphics.Color
        get() {
            return when (this) {
                RED -> com.badlogic.gdx.graphics.Color.RED
                YELLOW -> com.badlogic.gdx.graphics.Color.YELLOW
                GREEN -> com.badlogic.gdx.graphics.Color.GREEN
                MAGENTA -> com.badlogic.gdx.graphics.Color.MAGENTA
                BLUE -> com.badlogic.gdx.graphics.Color.BLUE
                CYAN -> com.badlogic.gdx.graphics.Color.CYAN
            }
        }

    companion object {

        fun getRandomExcept(colors: List<Color>): Color {
            var colorNum: Int
            while (true) {
                colorNum = (Math.random() * 6.0).toInt()
                when (colorNum) {
                    0 -> if (!colors.contains(Color.RED)) {
                        return RED
                    }
                    1 -> if (!colors.contains(Color.YELLOW)) {
                        return YELLOW
                    }
                    2 -> if (!colors.contains(Color.GREEN)) {
                        return GREEN
                    }
                    3 -> if (!colors.contains(Color.MAGENTA)) {
                        return MAGENTA
                    }
                    4 -> if (!colors.contains(Color.BLUE)) {
                        return BLUE
                    }
                    5 -> if (!colors.contains(Color.CYAN)) {
                        return CYAN
                    }
                    else -> throw IllegalStateException()
                }
            }
        }

        fun getAngleBetween(c1: Color, c2: Color): Float {
            val dif = c1.ordinal - c2.ordinal
            return (60 * dif).toFloat()
        }
    }
}
